src/lib/UXDController.ts

```diff
@@ -142,44 +156,50 @@ export class UXDController {
   }: {
     contractAddress: string;
     spender: string;
-    amount: BigNumber;
+    amount: number;
     signer: Signer;
   }): Promise<ContractTransaction> {
-    return ERC20__factory.connect(contractAddress, signer).connect(signer).approve(spender, amount);
+    const ethAmount = ethers.utils.parseEther(amount.toString());
+    return ERC20__factory.connect(contractAddress, signer).connect(signer).approve(spender, ethAmount);
   }
 
-  public allowance({
+  public async allowance({
     contractAddress,
     account,
     spender,
   }: {
     contractAddress: string;
     account: string;
     spender: string;
-  }): Promise<BigNumber> {
-    return ERC20__factory.connect(contractAddress, this.provider).allowance(account, spender);
+  }): Promise<number> {
+    const allowance = await ERC20__factory.connect(contractAddress, this.provider).allowance(account, spender);
+    return Number(ethers.utils.formatEther(allowance));
   }
 
-  public tokenBalance({
+  public async tokenBalance({
     contractAddress,
     account,
   }: {
     contractAddress: string;
     account: string;
-  }): Promise<BigNumber> {
-    return ERC20__factory.connect(contractAddress, this.provider).balanceOf(account);
+  }): Promise<number> {
+    const balance = await ERC20__factory.connect(contractAddress, this.provider).balanceOf(account);
+    return Number(ethers.utils.formatEther(balance));
   }
 
-  public uxdTotalSupply(): Promise<BigNumber> {
-    return this.uxdContract.totalSupply();
+  public async uxdTotalSupply(): Promise<number> {
```

Comment:
`getTotalRedeemableSupply` ?

Reply:
This is the total supply that has been minted, has nothing to do with redeeming. Every token on ETH has a current total supply and this is what this represents.

Reply:
The total UXD in circulation, or the total that has ever been minted, or the maximum supply?

We usually call UXD the "redeemable" (because every single UXD in circulation can always be redeemed)

Reply:
On coingecko any coin/token has a total supply, which is the total amount of the token currently in existence. This is standardized in the ERC-20 token standard, and notice the code is just pulling from the `totalSupply()` function of the token. In the UXD context, it could coincidentally be the amount redeemable, but this function does not say anything about redeeming.
The standard: https://eips.ethereum.org/EIPS/eip-20#methods

Reply:
Yeah so this is the redeemable (UXD) circulating supply.

This method is called `getRedeemableMintCirculatingSupply` in the Solana uxd-client.